src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -228,17 +228,11 @@ public async Task Skip(long len, bool async)
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public sbyte ReadSByte()
-        {
-            Debug.Assert(sizeof(sbyte) <= ReadBytesLeft);
-            return (sbyte)Buffer[ReadPosition++];
-        }
+            => Read<sbyte>();
```

Comment:
It seems a bit silly to have these wrappers if all they do it call `Read<T>()`, why not just remove them and have callers call `Read<T>()` directly?

Otherwise as a nit, we can at least put this on the same line as the function above.

Reply:
In my case you will have a common exception for all overflows. I think that it's very important.